ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op_width_sharded_program_factory.cpp

```diff
@@ -148,11 +148,19 @@ operation::ProgramWithCallbacks multi_core_optimized_conv_width_sharded_v2_impl(
     // TODO: Can conv_act_c_blocks be same as num_blocks_act_w?
     auto shard_shape = a.shard_spec().value().shape;
 
+    CoreRangeSet input_cores = a.memory_config().shard_spec.value().grid;
+    CoreRangeSet output_cores = output.memory_config().shard_spec.value().grid;
+    CoreRangeSet all_cores = output.memory_config().shard_spec.value().grid;
+    if(input_cores.num_cores()>output_cores.num_cores()){
+        all_cores = a.memory_config().shard_spec.value().grid;
```

Comment:
a.memory_config().shard_spec.value().grid == input_cores rigth?
Pls format the code